db/migrate/20220220190037_devise_token_auth_create_users.rb

```diff
@@ -53,5 +53,6 @@ def change
     add_index :users, :reset_password_token, unique: true
     add_index :users, :confirmation_token,   unique: true
     add_index :users, :unlock_token,         unique: true
+    add_index :users, :username,             unique: true
```

Comment:
we have decision to make validations on model level

Reply:
I don't quite understand, can you rephrase?

Reply:
as I understand the issue, we need to leave validations on the model level as you did before the last commit. but I'm not sure and ask help

Reply:
ok, thanks for your review!